bigtable/bttest/grpc_error_injection.go

```diff
@@ -0,0 +1,307 @@
+/*
+Copyright 2021 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+This file provides interceptors to hook into grpc streaming calls
+and inject latency and errors into cbtemulator for testing
+*/
+package bttest
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"strconv"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var (
+	validStreamMethodSuffixes = []string{"ReadRows", "MutateRows"}
+)
+
+/*
+   Creates interceptors inject latency or errors into cbtemulator
+*/
+type EmulatorInterceptor struct {
+	LatencyTargets       latencyTargets
+	GrpcErrorCodeTargets grpcErrorCodeTargets
+}
+
+func (esib *EmulatorInterceptor) StreamInterceptor() grpc.ServerOption {
```

Comment:
Small nit: What is `esib`? Perhaps `ei` or just `e` would be adequate? Cf. https://github.com/golang/go/wiki/CodeReviewComments#receiver-names